File: app/src/main/java/com/spidchenko/week2task/db/dao/SyncImageDao.java
package com.spidchenko.week2task.db.dao;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import com.spidchenko.week2task.db.models.SyncImage;

import java.util.List;

@Dao
public interface SyncImageDao {

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    long addSyncImage(SyncImage syncImage);

    @Delete
    void deleteSyncImage(SyncImage syncImage);

    @Query("SELECT * FROM syncImages")
    LiveData<List<SyncImage>> getAllSyncImages();
}

File: app/src/main/java/com/spidchenko/week2task/db/dao/SyncImageDao.kt
package com.spidchenko.week2task.db.dao

import androidx.room.Dao
import androidx.room.OnConflictStrategy
import com.spidchenko.week2task.db.models.SyncImage
import androidx.lifecycle.LiveData
import androidx.room.Insert
import androidx.room.Query

@Dao
interface SyncImageDao {
    @Insert(onConflict = OnConflictStrategy.IGNORE)
    fun addSyncImage(syncImage: SyncImage?): Long

    @Query("DELETE FROM syncImages WHERE url LIKE :url")
    fun deleteSyncImage(url: String?)

    @get:Query("SELECT * FROM syncImages")
    val allImages: LiveData<List<SyncImage?>?>?
}